src/solve3.F

```diff
@@ -951,67 +951,47 @@ subroutine solve3(nstep,rbufsz,num_soil_layers,                  &
         if (prcl_droplet.ge.1) then
 
           !$acc parallel default(present) private(i,j,k)
-          !$acc loop gang vector collapse(3)
-          do k=kb,ke+1
-          do j=jb,je
-          do i=ib,ie
-            wten(i,j,k) = 0.0
-          enddo
-          enddo
-          enddo
-          
-          !$acc loop gang vector collapse(3)
-          do k=kb,ke
-          do j=jb,je
-          do i=ib,ie+1
-            uten(i,j,k) = 0.0
-          enddo
-          enddo
-          enddo
-
-          !$acc loop gang vector collapse(3)
-          do k=kb,ke
-          do j=jb,je+1
-          do i=ib,ie
-            vten(i,j,k) = 0.0
-          enddo
-          enddo
-          enddo
-
           !$acc loop gang vector collapse(3)
           do k=kb,ke
           do j=jb,je
           do i=ib,ie
             dpten(i,j,k,1) = 0.0
             dpten(i,j,k,2) = 0.0
+            dpten(i,j,k,3) = 0.0
+            dpten(i,j,k,4) = 0.0
+            dpten(i,j,k,5) = 0.0
+            dpten(i,j,k,6) = 0.0
           enddo
           enddo
           enddo
           !$acc end parallel
 
-          call   droplet_driver(dt,mtime,xh,uh,ruh,xf,yh,vh,rvh,yf,zh,mh,rmh,zf,mf,zs,    &
+          call   droplet_driver(dt,dbldt,mtime,xh,uh,ruh,xf,yh,vh,rvh,yf,zh,mh,rmh,zf,mf,zs, &
                                sigma,sigmaf,zntmp,rho,rru,rrv,rrw,s10,pdata,              &
-                               tha,qa,th0,pi0,ppi,prs,                                    &
+                               tha,qa,th0,pi0,ppi,prs,dum1,                               &
                                pw1,pw2,pe1,pe2,ps1,ps2,pn1,pn2,                           &
                                nw1,nw2,ne1,ne2,sw1,sw2,se1,se2,reqs_p,                    &
                                sw31,sw32,se31,se32,ss31,ss32,sn31,sn32,                   &
                                n3w1,n3w2,n3e1,n3e2,s3w1,s3w2,s3e1,s3e2,reqs_s,            &
-                               uten,vten,wten,dpten(ib,jb,kb,1),dpten(ib,jb,kb,2),        &
+                               dpten(ib,jb,kb,3),dpten(ib,jb,kb,4),dpten(ib,jb,kb,5),dpten(ib,jb,kb,1),dpten(ib,jb,kb,2),dpten(ib,jb,kb,6),        &
                                pdata_locind)
           if(timestats.ge.1) time_parcels=time_parcels+mytime()
 #ifdef MPI
           call sync()
           if(timestats.ge.1) time_lb=time_lb+mytime()
           call comm_1s_tend_halo_GPU(dpten(ib,jb,kb,1))
           call comm_1s_tend_halo_GPU(dpten(ib,jb,kb,2))
+          call comm_1u_tend_halo_GPU(dpten(ib,jb,kb,3))
+          call comm_1v_tend_halo_GPU(dpten(ib,jb,kb,4))
+          call comm_1s_tend_halo_GPU(dpten(ib,jb,kb,5))
 #else
           call bcs_tend_halo(dpten(ib,jb,kb,1))
           call bcs_tend_halo(dpten(ib,jb,kb,2))
 #endif
 #ifdef MPI
           if(timestats.ge.1) time_dpc=time_dpc+mytime()
 #endif
-          if( nout3d.ge.2 )then
+          if( nout3d.ge.6 )then
```

Comment:
Right -- it's set at the top of cm1.F. It's typically set to 0. The user can change this in order to output 3D variables if they want to analyze them. Otherwise the code ignores them.

Reply:
Oh, I see. This is just for output purpose, not affecting the results?

Reply:
Correct